### PR TITLE
fix: store-1343 - add underlyingNetwork to bitfinex and huobi

### PR DESCRIPTION
**Pull request by NeOMakinG**
Underlying networks is needed in order to avoid pushing methods to the `network` prop which could be used to withdraw

Actually we didn't modify the actual behavior of the exchange but added a key in order to map the network with our internal network, these keys aren't internal keys but external keys, so it's not mixed up with our internal mapping

## Huobi

My dear Huobi, thanks a lot for this hard time managing to map your networks properly, in order to achieve it I had to:

- Generate methods with a pattern (currencyName + network, network + currencyName (example: usdcerc20 or erc20usdc)
- Add it to a mapping object with method as a key, `ecid` as a value
- consume it to display the underlying network

### CODE REVIEW

File: ts/src/bitfinex2.ts [hunk 1]
```diff
@@ -778,19 +779,36 @@ export default class bitfinex2 extends Exchange {
...
+            'TETHEREUE': 'ETH',
+            'TETHERCNHTX': 'ETH',
+            'TETHERCNHTE': 'ETH',
+            'TETHERUSDTZK': 'ZK2',
+            'XAUT': 'ETH',
+            'CNHT': 'ETH',
+            'EUT': 'BSC',
+            'MXNT': 'ETH',
+            'NEOGAS': 'NEO',
+            'MATICM': 'NEO',
```
Comment (neeeekitos) [inline ts/src/bitfinex2.ts:811]: I think it might be an error

Comment (NeOMakinG) [reply ts/src/bitfinex2.ts]: Yeah it was probably a wrong copy pasta as `MATICM` is mapped to polygon, sorry

File: ts/src/huobi.ts [hunk 2]
```diff
@@ -2876,10 +2961,9 @@ export default class huobi extends Exchange {
             for (let j = 0; j < chains.length; j++) {
                 const chainEntry = chains[j];
                 const uniqueChainId = this.safeString (chainEntry, 'chain'); // i.e. usdterc20, trc20usdt ...
-                const title = this.safeString (chainEntry, 'displayName');
-                this.options['networkChainIdsByNames'][code][title] = uniqueChainId;
-                this.options['networkNamesByChainIds'][uniqueChainId] = title;
-                const networkCode = this.networkIdToCode (title, code);
+                const displayName = this.safeString (chainEntry, 'displayName');
+                // this.options['networkNamesByChainIds'][uniqueChainId] = title;
+                // const networkCode = this.networkIdToCode (title, code);
```
Comment (neeeekitos) [inline ts/src/huobi.ts:2966]: We can remove it

File: ts/src/huobi.ts [hunk 3]
```diff
@@ -2809,6 +2816,82 @@ export default class huobi extends Exchange {
...
+            'mbx': 'klay',
+            'npt': 'klay',
+            'we': 'klay',
+            'well': 'glmr',
+            'wmt': 'ada',
+            'roco': 'cchain',
+            'fio': 'cchain',
+            'hec': 'cchain',
+            'gmx': 'cchain', // Not sure, as it's tagged as `cchainerc20`
+            'xeta': 'cchain', // Not sure, as it's tagged as `cchainerc20`
```
Comment (neeeekitos) [inline ts/src/huobi.ts:2885]: Let's make sure if it's correct (if yes, we can remove those comments)

**Review by neeeekitos — COMMENTED** [review]
Great job 👍

**Pull request merged**